app/controllers/posts_controller.rb

```diff
@@ -1,10 +1,34 @@
 class PostsController < ApplicationController
   def index
-    # @posts = Post.all
     @users = User.find(params[:user_id])
   end
 
   def show
     @post = Post.find(params[:id])
+    @comments = Comment.where(post_id: params[:id])
+  end
```

Comment:
- [ ] Good :+1: job :100: trying to solve the `n + 1` problem. But I think it would be better if you solve it in your `controllers` instead of `views`. This is because `controllers` handle database queries and other data retrieval logic, so addressing the N+1 problem in the controller ensures that only the necessary data is retrieved, reducing the number of queries and improving performance. You can refer this given **[article](https://dev.to/junko911/rails-n-1-queries-and-eager-loading-10eh)** from the project requirements or **[this](https://medium.com/idopterlabs/solving-the-n-1-problem-in-rails-b5ea677e78e0)** or especially take a look on this **[template example](https://gist.github.com/tresorsawasawa/af9c2d060489ab0d2c42391ea7e4dd54)**. Once you fix the above issue, I kindly suggest you to add new screenshots in your PR description. I know you can do that :muscle: :muscle: